File: mirador/add_manifests.js
var fs = require('fs');
var pretty = require('pretty');

function buildDocument(miradorObj, identifier) {
  var doc = pretty('<!DOCTYPE html>'
       + '<html>' + '<head>' + '<title>' + identifier + '</title>'
       + '<meta charset="UTF-8">' + '<meta name="viewport" content="width=device-width, initial-scale=1.0">'
       + '<link rel="stylesheet" type="text/css" href="build/mirador/css/mirador-combined.css">'
       + '<link rel="stylesheet" type="text/css" href="build/mirador/css/mirador-bc.css">'
       + '<script src="build/mirador/mirador.js"></script>' + '</head>' + '<body>'
       + '<div id="viewer"></div>' + '<script type="text/javascript">'
       + '$(function() { Mirador(' + miradorObj + '); });' + '</script>' + '</body>' + '</html>');

  fs.writeFile(identifier, doc, (err) => {
    if (err) throw err;
    console.log(`Saved ${identifier} to file`);
  });
}

function readLines(input, func) {
  var remaining = '';

  input.on('data', function(data) {
    remaining += data;
    var index = remaining.indexOf('\n');
    while (index > -1) {
      var line = remaining.substring(0, index);
      remaining = remaining.substring(index + 1);
      func(line);
      index = remaining.indexOf('\n');
    }
  });

  input.on('end', function() {
    if (remaining.length > 0) {
      func(remaining);
    }
  });
}

function loadManifestData(data) {
  var mirador = {
    "id": "viewer",
    "mainMenuSettings": {
      "buttons": {
        "bookmark": false,
        "fullScreenViewer": false
      },
      "userLogo": {
        "label": "Boston College Library",
        "attributes": {
          "id": "bc-logo",
          "href": "https://library.bc.edu"
        }
      }
    },
    "data": [],
    "windowObjects": []
  };
  var location = {
    "manifestUri": data,
    "location": "Boston College"
  };
  var loaded = {
    "loadedManifest": data,
    "viewType": "ImageView"
  };

  mirador["data"].push(location);
  mirador["windowObjects"].push(loaded);

  var identifier = data.split('/').pop().slice(0, -5);
  var miradorObj = JSON.stringify(mirador);

  buildDocument(miradorObj, identifier);
}

var input = fs.createReadStream('./manifests.txt');
readLines(input, loadManifestData);

File: mirador/add_manifests.rb
require 'json'
require 'pathname'

def handle_input(file)
    if ARGV.empty?
    puts "Error: no argument supplied"
    puts "Usage: ruby add_manifests.rb some_iiif_manifest.json\n"
    exit
  elsif ARGV.length > 1
    puts "Error: takes only one argument\n"
    puts "Usage: ruby add_manifests.rb some_iiif_manifest.json\n"
    exit
  elsif json_file.slice(-5, 5) != '.json'
    puts "Error: input file must be JSON\n"
    puts "Usage: ruby add_manifests.rb some_iiif_manifest.json\n"
    exit
  elsif !Pathname(file).exist?
    puts "Error: #{file} could not be found"
    exit
  end
end

def parse_manifest_file(file)
  f = File.read(file)
  JSON.parse(f)
end

def construct_mirador_object(file)
  manifest_hash = parse_manifest_file(file)
  manifest_uri = "https://library.bc.edu/iiif/manifests/#{identifier}"
  handle = manifest_hash

  mirador = {
    "id": "viewer",
    "mainMenuSettings": {
      "buttons": {
        "bookmark": false,
        "fullScreenViewer": false
      },
      "userButtons": [{
        "label": "View Library Record",
        "iconClass": "fa fa-external-link",
        "attributes": {}
      }],
      "userLogo": {
        "label": "Boston College Library",
        "attributes": {
          "id": "bc-logo",
          "href": "https://library.bc.edu"
        }
      }
    },
    "data": [],
    "windowObjects": []
  }
  location = { "manifestUri": manifest_uri, "location": "Boston College" }
  loaded = { "loadedManifest": manifest_uri, "viewType": "ImageView" }
  hdl_button = { "class": "handle", "href": handle }

  mirador[:data].push(location)
  mirador[:windowObjects].push(loaded)
  mirador[:mainMenuSettings][:userButtons][0][:attributes] = hdl_button
end

def build_document(mirador_object)
  identifer = 
  doc = <<-EOF
<!DOCTYPE html>
<html>

<head>
  <title>#{identifier}</title>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <link rel="stylesheet" type="text/css" href="build/mirador/css/mirador-combined.css"></link>
  <link rel="stylesheet" type="text/css" href="build/mirador/css/mirador-bc.css"></link>
  <script src="build/mirador/mirador.js"></script>
</head>

<body>
  <div id="viewer"></div>
  <script type="text/javascript">
    $(function() {
      Mirador(#{mirador_object.to_json});
  </script>
</body>

</html>
  EOF
end

input_file = ARGV[0]
handle_input(input_file)